torchvision/transforms/v2/_auto_augment.py

```diff
@@ -94,6 +94,7 @@ def _apply_image_or_video_transform(
         interpolation: Union[InterpolationMode, int],
         fill: Dict[Union[Type, str], _FillTypeJIT],
     ) -> ImageOrVideo:
+        image = cast(torch.Tensor, image)
```

Comment:
Maybe add a comment here that this cast is wrong and we are only doing this for JIT.